nifi-nar-bundles/nifi-parquet-bundle/nifi-parquet-processors/src/main/java/org/apache/nifi/parquet/utils/ParquetUtils.java

```diff
@@ -168,9 +168,11 @@ public static ParquetConfig createParquetConfig(final PropertyContext context, f
         final ParquetFileWriter.Mode mode = overwrite ? ParquetFileWriter.Mode.OVERWRITE : ParquetFileWriter.Mode.CREATE;
         parquetConfig.setWriterMode(mode);
 
-        final String compressionTypeValue = context.getProperty(ParquetUtils.COMPRESSION_TYPE).getValue();
-        final CompressionCodecName codecName = CompressionCodecName.valueOf(compressionTypeValue);
-        parquetConfig.setCompressionCodec(codecName);
+        if(context.getProperty(ParquetUtils.COMPRESSION_TYPE).isSet()) {
+            final String compressionTypeValue = context.getProperty(ParquetUtils.COMPRESSION_TYPE).getValue();
+            final CompressionCodecName codecName = CompressionCodecName.valueOf(compressionTypeValue);
+            parquetConfig.setCompressionCodec(codecName);
+        }
```

Comment:
The COMPRESSION_TYPE PropertyDescriptor has a list of allowable values and it also has a default value using the first value which is UNCOMPRESSED, anything with a default value should always return a value so that is why the code didn't check isSet() before.

I'm not terribly offended adding this as an extra precaution, but I am concerned that something else broke that now makes the default value not return.

Reply:
There is no COMPRESSION_TYPE PropertyDescriptor in the ParquetReader controller service which is causing the NPE initially discussed in #4537.

Reply:
That is correct, but we always called `ParquetUtils createParquetConfig`  from both the reader and writer before and it worked, so I'm confused why now we have to check isSet().

The JIRA mentioned https://issues.apache.org/jira/browse/NIFI-7635  so maybe that fix now makes it such that if you call `context.getProperty(ParquetUtils.COMPRESSION_TYPE).getValue()` and the component doesn't have that property then you no longer get the default value. Before you did get the default value.

Reply:
This is most likely because of this change: https://github.com/apache/nifi/commit/4f11e3626093d3090f97c0efc5e229d83b6006e4#diff-782335ecee68f6939c3724dba3983d3d
That's definitely a good question though: did this impact only this Parquet controller service or did it impact other components? custom code/NARs/etc?

Reply:
PR #4537 aimed to allow previous functionality to work as it did in previous versions of NiFi. However, the resolution order changed to also address NIFI-7635. In that PR, the `resolvedValue` was determined in this order:
- configuredValue
- default value from the ComponentNode
- default value from the PropertyDescriptor

Whereas in earlier versions of NiFi (pre 1.12.0), the resolution was done in this order:
- configuredValue
- default value from the PropertyDescriptor
- default value from the ComponentNode